clang/test/SemaSYCL/sycl-device-intel-fpga-no-global-work-offset.cpp

```diff
@@ -0,0 +1,61 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsyntax-only -ast-dump -verify -pedantic %s | FileCheck %s
+
+// Test that checks template parameter support for 'no_global_work_offset' attribute on sycl device.
+
+// Test that checks wrong template instantiation.
+template <typename Ty>
+[[intel::no_global_work_offset(Ty{})]] void func() {}
+
+struct S {};
+// expected-error@+2{{template specialization requires 'template<>'}}
+// expected-error@+1{{C++ requires a type specifier for all declarations}}
+func<S>();
```

Comment:
I apologize, my suggestion was unclear. I intended for you to instantiate the function template to ensure that the type was checked properly when instantiating from the template definition. You should replace the call here to `func<>()` with:
```
void bar() {
  func<S>();
}
```

Reply:
Done